appserver/web/web-core/src/main/java/org/apache/catalina/session/PersistentManagerBase.java

```diff
@@ -600,6 +600,17 @@ protected void processExpires() {
         final List<Session> sessions = findSessions();
         for (final Session session1 : sessions) {
             StandardSession session = (StandardSession) session1;
+            //verify if session also is on the store and compare last access time
+            StandardSession sessionFromStore = null;
+            try {
+                sessionFromStore = (StandardSession) swapIn(session.getId());
+            } catch (IOException e) {
+                log.log(Level.WARNING, "Session from Store not available", e);
```

Comment:
Is this correct? Just eyeballing it right now, but it seems more like an IOException isn't thrown if the session isn't available but if there's an issue attempting to retrieve from the store e.g. security manager enabled.

I'd suggest changing the message to something along the lines of "Caught exception attempting to swap in session from store", or reviewing the code path and seeing if it makes sense to just throw the IOException and let something else handle it.

Reply:
Having slept on it, throwing an exception probably isn't correct since this method is attempting to expire stuff, so probably better to just review the message and log level (warning might be too high? I'm undecided)

Reply:
I already change message and also the level to info